src/sentry/rules/processing/processor.py

```diff
@@ -302,6 +334,42 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
                 )
                 return
 
+        predicate_func = get_match_function(condition_match)
+        if not predicate_func:
+            log_string = f"Unsupported condition_match {condition_match} for rule {rule.id}"
+            logger.error(
+                log_string,
+                filter_match,
+                rule.id,
+                extra={**logging_details},
+            )
+            return
+        skip_slow_conditions_check = False
+        if condition_list:
```

Comment:
If we replace `condition_list` in this if statement with `fast_conditions or slow_conditions` can we get rid of the if statement outside? This code is getting hard to follow and will be hard to maintain as is, so I'd really like to simplify

Reply:
You mean get rid of the `if not predicate_func` and logging?

Reply:
I just added a test for that, we need to `return` `if not predicate_func` otherwise the alert would fire if someone (probably via the API) makes a rule with an unsupported action match (unless the serializer protects against that, but the way this is written it sounds like it doesn't).

Reply:
No, the `if slow_conditions and process_slow_conditions_later and not skip_slow_conditions_check:` after this if statement.